source/platforms/GitLab.ts

```diff
@@ -99,28 +100,49 @@ class GitLab implements Platform {
   updateOrCreateComment = async (dangerID: string, newComment: string): Promise<string> => {
     d("updateOrCreateComment", { dangerID, newComment })
 
-    //Even when we don't set danger to create threads, we still need to delete them if someone answered to a single
-    // comment created by danger, resulting in a discussion/thread. Otherwise we are left with dangling comments
-    // that will most likely have no meaning out of context.
-    const discussions = await this.getDangerDiscussions(dangerID)
-    const firstDiscussion = discussions.shift()
-    const existingNote = firstDiscussion?.notes[0]
-    // Delete all notes from all other danger discussions (discussions cannot be deleted as a whole):
-    await this.deleteNotes(this.reduceNotesFromDiscussions(discussions)) //delete the rest
-
-    let newOrUpdatedNote: GitLabNote
+    if (useThreads()) {
+      const discussions = await this.getDangerDiscussions(dangerID)
+      const firstDiscussion = discussions.shift()
+      const existingNote = firstDiscussion?.notes[0]
+      // Delete all notes from all other danger discussions (discussions cannot be deleted as a whole):
+      await this.deleteNotes(this.reduceNotesFromDiscussions(discussions)) //delete the rest
+
+      let newOrUpdatedNote: GitLabNote
+
+      if (existingNote) {
+        // update the existing comment
+        newOrUpdatedNote = await this.api.updateMergeRequestNote(existingNote.id, newComment)
+      } else {
+        // create a new comment
+        newOrUpdatedNote = await this.createComment(newComment)
+      }
 
-    if (existingNote) {
-      // update the existing comment
-      newOrUpdatedNote = await this.api.updateMergeRequestNote(existingNote.id, newComment)
+      // create URL from note
+      // "https://gitlab.com/group/project/merge_requests/154#note_132143425"
+      return `${this.api.mergeRequestURL}#note_${newOrUpdatedNote.id}`
     } else {
-      // create a new comment
-      newOrUpdatedNote = await this.createComment(newComment)
+      const notes: GitLabNote[] = await this.getMainDangerNotes(dangerID)
```

Comment:
This is copied from the code that existed before the change in https://github.com/danger/danger-js/pull/1331